R/sits_kohonen.R

```diff
@@ -1,92 +1,770 @@
+
+    confusion.tb<- dplyr::select(
+        Confusion_and_neighboor,
+        id_sample,
+        original_label,
+        cluster_label,
+        P_sample=percentage,
+        label_neighbor_neuron,
+        P_Neighbor=(freq)
+    )%>%dplyr::arrange(desc(P_sample))
+
+
+    Matching_and_neighboor <-
+        general_vinicity %>% dplyr::inner_join(matching_between_samples.tb)
+
+    matching.tb<- dplyr::select(
+        Matching_and_neighboor,
+        id_sample,
+        original_label,
+        cluster_label,
+        P_sample=percentage,
+        label_neighbor_neuron,
+        P_Neighbor=(freq)
+    )%>%dplyr::arrange(desc(P_sample))
+
 
-    #verifies if data.tb has the same size of kohonen.obj
-    ensurer::ensure_that(data.tb, (nrow(.) == length(kohonen.obj$unit.classif)),
-                         err_desc = "sits_kohonen_cluster: you must provide the same data set used to cluster data.")
+    evaluated_sample <-  structure(list(
+        table_sample_neuron= table_sample_neuron,
+        metrics_by_samples = info_sample_cluster.tb,
+        samples.tb = samples_new_label
+    ),
+    class = "sits")
+}
+
+
+#' @title Get the neighbor of neurons
+#' @name .sits_neighbor_neurons
+#' @author Lorena Santos, \email{lorena.santos@@inpe.br}
+#'
+#' @description This function create a table contained the information about all
+#' neighbor of one neuron of radius 1
+#'
+#' @param class_vector  a vector contained the labels of each neuron
+#' @param koh an object kohonen
+#' @param radius radius of neigbourhood of each neuron
+#'
+#' @return returns a SITS tibble with informations about the vinicity of each neuron
+
+.sits_neighbor_neurons <- function (class_vector,koh,radius=1)
+{
+    neuron_vicinity.tb <- as_tibble()
+    grid_size <- dim(koh$grid$pts)[1]
+    neuron=1
+    for (neuron in 1:grid_size)
+    {
+        neurons_neighbors<-which(unit.distances(koh$grid)[,neuron] == radius)
+        class_neighbors<-class_vector[neurons_neighbors]
+
+        count_neighbors<-table(class_neighbors)
+        result_vizinhos_temporary.tb <- tibble::as_tibble(list(
+            id_neuron= as.integer(neuron),
+            neuron_label= as.character(class_vector[neuron]),
+            label_neighbor_neuron =names(count_neighbors),
+            F_Neighbor = as.integer(count_neighbors),
+            P_Neighbor  = as.numeric(prop.table(count_neighbors))
+        ))
+
+        neuron_vicinity.tb <- rbind(neuron_vicinity.tb, result_vizinhos_temporary.tb)
+
+    }
+    return (neuron_vicinity.tb)
+
+}
+
+#' @title Create new groups from kohonen maps
+#' @name sits_subgroup
+#' @author Lorena Santos, \email{lorena.santos@@inpe.br}
+#'
+#' @description Create new groups to identify variations in a same group
+#'
+#' @param koh  An object with informations about kohonen clustering and samples.
+#' @return returns a SITS tibble with subgroups generated by hierarchical clustering.
+#' @export
+sits_subgroup<-function(koh, plot_subgroups = TRUE)
```

Comment:
It is never good to have plot and computation in a single function. Better to split it in two (and test them as well).